docs/DeveloperGuide.md

```diff
@@ -1,11 +1,523 @@
+    * `TaskManager` to facilitate the management of `Tasks` that are broken down from the `Project`.
+##### Members Operations
+* `ProjectMembers` facilitate the management of multiple `Members` instances and is dependant on the `Project` that initialises it.
+    * Creation of `Member` adds an entry to `ProjectMembers`.
+    * Deletion of `Member` removes an entry from `ProjectMembers`.
+* `Member` contain the Member's user ID.
+
+##### Task Operations
+* `TaskManager` facilitate the management of multiple `Task` instances and is dependant on the `Project` that initialises it.
+    * Creation of `Task` adds an entry to `TaskManager`.
+    * Deletion of `Task` removes an entry from `TaskManager`.
+* `Task` contain necessary information about the task such as:
+    * Task ID
+    * Task Title
+    * Task Description
+    * Task Priority
+    * Task Completion Status
+* `Task` contain two additional ArrayList that are initialise upon its creation:
+    * ArrayList of Member's user ID to keep track `Members` who are assigned to work on the task.
+    * ArrayList of Sprint ID to keep track `Sprints` that the task are allocated to.
+
+##### Sprint Operations
+* `SprintManager` will facilitate the management of multiple `Sprint` instances and is dependant on the `Project` that initialises it.
+    * Creation of `Sprint` adds an entry to `SprintManager`.
+* `Sprint` contain necessary information about the iteration such as:
+    * Sprint ID
+    * Sprint Goal
+    * Sprint Start Date
+    * Sprint End Date
+* `Sprint` contain one additional ArrayList that are initialise upon its creation:
+    * ArrayList of Task IDs to keep track `Tasks` that are allocated to the `Sprint`.
+### Storage Component
+![Figure X: Simplified class diagram for Storage Component, Model and json.simple](./image/developerguide/storagecomponent.png "Storage Component UML") 
+API: [StorageManager.java]( https://github.com/AY2021S1-CS2113T-F11-4/tp/tree/master/src/main/java/seedu/duke/storage/StorageManager.java)  
+The Storage component is using the JavaScript Object Notation (JSON) to save the data. The library used for serialising and deserializing the data is _json.simple 3.1.1_ by **Clifton Labs**.  
+As shown in the diagram above, `JsonableObject` and `JsonableArray` are interfaces which inherits the `Jsonable` interface. The following model class inherits only one of the two interfaces:  
+- ProjectManager  
+- Project  
+- SprintManager  
+- TaskManager  
+- ProjectMembers  
+- Sprint  
+- Task  
+- Member  
+
+This requires the model classes to implement two methods required for JSON serialisation and deserialisation:  
+- `toJson()`: Contains logic required to convert the model object into JSON string.  
+- `fromJson()`: Contains logic required to convert JSON object into its respective model class.    
+
+## Implementation
+### Project
+#### Create Project
+![Figure X: Sequence diagram of CreateProjectCommand](./image/developerguide/createProjectSequenceDiagram.png
```

Comment:
Some of the methods called included the dot sign such as "projectManager.addProject()", perhaps it should have been "addProject()" instead.